### ISSUE TITLE
On-reaction deletion of bot messages

### ISSUE THREAD

**Issue by pomegranategurk** [gurkult/gurkbot#89]
Feature: Say when **someone sends !gh source**, the bot will **send the github link like normal.** But **it will react to the message** (The message the bot sent) with a (example) trash can. When the **original sender of the message reacts to the message with the trash can, it will delete the github link**.

**Comment by gustavwilliam** [comment]
We should definitely have this. However, I'd like it to be the default for the bot embeds, instead of being manually applied on every command. There should aslo be a setting to disable the reaction for that specific embed. I wouldn't mind using decorators or any other implementation that you could figure out.

**Comment by pomegranategurk** [comment]
(this is my first issue so i might be a little slow) Are you saying that only embeds have a deletion feature?

**Comment by gustavwilliam** [comment]
Nope. I think this should apply to all bot outputs, but we almost exclusively use embeds.

**Comment by pomegranategurk** [comment]
Cool. When you were talking about decorators are you saying that in the bots code say we had `await channel.send` we would put (example) `@deletable` before the message is sent?

**Comment by gustavwilliam** [comment]
Yep, that would be the idea. However, I'm not sure what the best way forward would be.

**Comment by codephile1221** [comment]
Yes, I think we exclusively use embeds for bot outputs. However, any output sent in a normal message form will anyway be short and will not take much screen space, and hence not warrant the need for it to be deleted in most cases.

**Comment by codephile1221** [comment]
I think #83  will handle the deletable embed issue. However, there has been no correspondence by the author on the issue lately :angry:  .

**Comment by pomegranategurk** [comment]
Sorry for not responding, I had to eat lunch.
But I will look into ways to possibly implement this. And even we don’t have a need for it now, it might come in handy down the road.

**Comment by pomegranategurk** [comment]
I might not have time to implement this as classes are getting weird since a bunch of kids have gotten quarantined. (And I don't know what i'm doing, so I'd have to spend extra time googling.)
